tgui/packages/tgui/components/DropdownWithSwitchButtons.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * @file
+ * @license MIT
+ */
+
+import { classes } from 'common/react';
+import { Box } from './Box';
+import { Icon } from './Icon';
+import { Dropdown } from './Dropdown';
+import { Button } from './Button';
+import { Stack } from './Stack';
+
+export class DropdownWithSwitchButtons extends Dropdown {
```

Comment:
You should never extend a tgui component. Have this compose using a `Dropdown` inside it instead.

Reply:
This code also looks *incredibly* complicated given that what this is doing sounds as simple as "make a dropdown, have two buttons that set its value"